lock-api/src/test/java/com/palantir/lock/client/UnreliableTimelockClientTest.java

```diff
@@ -0,0 +1,179 @@
+        verify(timeLockClient).getCommitTimestamp(1, LOCK_TOKEN);
+    }
+
+    @Test
+    public void verifyGetFreshTimestampsCallsDelegate() {
+        unreliableTimeLockService.getFreshTimestamps(1);
+        verify(timeLockClient).getFreshTimestamps(1);
+    }
+
+    @Test
+    public void verifyLockImmutableTimestampCallsDelegate() {
+        unreliableTimeLockService.lockImmutableTimestamp();
+        verify(timeLockClient).lockImmutableTimestamp();
+    }
+
+    @Test
+    public void verifyStartIdentifiedAtlasDbTransactionCallsDelegate() {
+        unreliableTimeLockService.startIdentifiedAtlasDbTransactionBatch(1);
+        verify(timeLockClient).startIdentifiedAtlasDbTransactionBatch(1);
+    }
+
+    @Test
+    public void verifyGetImmutableTimestampCallsDelegate() {
+        unreliableTimeLockService.getImmutableTimestamp();
+        verify(timeLockClient).getImmutableTimestamp();
+    }
+
+    @Test
+    public void verifyLockCallsDelegate() {
+        unreliableTimeLockService.lock(LOCK_REQUEST);
+        verify(timeLockClient).lock(LOCK_REQUEST);
+    }
+
+    @Test
+    public void lockMayUnlockImmediately() {
+        when(timeLockClient.lock(LOCK_REQUEST)).thenReturn(LockResponse.successful(LOCK_TOKEN));
+        new UnreliableTimeLockService(timeLockClient, createAlwaysBuggyFactory()).lock(LOCK_REQUEST);
+        verify(timeLockClient).unlock(Set.of(LOCK_TOKEN));
+    }
+
+    @Test
+    public void lockWithOptionMayUnlockImmediately() {
+        ClientLockingOptions lockingOptions = ClientLockingOptions.getDefault();
+        when(timeLockClient.lock(LOCK_REQUEST, lockingOptions)).thenReturn(LockResponse.successful(LOCK_TOKEN));
+        new UnreliableTimeLockService(timeLockClient, createAlwaysBuggyFactory()).lock(LOCK_REQUEST, lockingOptions);
+        verify(timeLockClient).unlock(Set.of(LOCK_TOKEN));
+    }
+
+    @Test
+    public void verifyLockWithOptionsCallsDelegate() {
+        unreliableTimeLockService.lock(LOCK_REQUEST, ClientLockingOptions.getDefault());
+        verify(timeLockClient).lock(LOCK_REQUEST, ClientLockingOptions.getDefault());
+    }
+
+    @Test
+    public void verifyWaitForLocksCallsDelegate() {
+        WaitForLocksRequest waitForLocksRequest = WaitForLocksRequest.of(Set.of(LOCK_DESCRIPTOR), 1);
+        unreliableTimeLockService.waitForLocks(waitForLocksRequest);
+        verify(timeLockClient).waitForLocks(waitForLocksRequest);
+    }
+
+    @Test
+    public void refreshLockLeasesMaybeUnlocksLocks() {
+        BuggifyFactory factory = mock(BuggifyFactory.class);
+        when(factory.maybe(anyDouble())).thenReturn(DefaultBuggify.INSTANCE, NoOpBuggify.INSTANCE);
+        new UnreliableTimeLockService(timeLockClient, factory)
+                .refreshLockLeases(new LinkedHashSet<>(List.of(LOCK_TOKEN, LOCK_TOKEN_2)));
+        verify(timeLockClient).refreshLockLeases(Set.of(LOCK_TOKEN_2));
+        verify(timeLockClient).unlock(Set.of(LOCK_TOKEN));
+    }
```

Comment:
nit: the extreme pedant in me wants to say for each method, the checks for "verify delegate is called" and "buggify does bad stuff" should be in a consistent ordering